### PR TITLE
Add perf8 integration

### PR DESCRIPTION
**Pull request by moxarth-elastic**
## Closes https://github.com/elastic/connectors-py/issues/140

Added psutil and memray from perf8 repo in ftest.

#### Pre-Review Checklist
- [x] Tested the changes locally

#### Changes Requiring Extra Attention

- [x] New external service dependencies added.

### CODE REVIEW

**Comment by tarekziade** [comment]
let's put this one on hold until `perf8` is public again

**Comment by tarekziade** [comment]
ok perf8 is back https://github.com/elastic/perf8

File: test-requirements.txt [hunk 1]
```diff
@@ -6,3 +6,4 @@ pytest>=7.1.2
 pytest-cov>=3.0.0
 pytest-asyncio>=0.19.0
 pytest-randomly>=3.12.0
+git+https://github.com/tarekziade/perf8@main#egg=perf8
```
Comment (tarekziade) [inline test-requirements.txt:9]: we can use 
```
git+https://github.com/elastic/perf8#egg=perf8
```

File: connectors/tests/ftest.sh [hunk 2]
```diff
@@ -14,12 +17,25 @@ sleep 30
...
+then
+    $ROOT_DIR/bin/perf8 --psutil --memray -c $ROOT_DIR/bin/elastic-ingest --one-sync --sync-now --debug
+
+elif [ $MEMPROFILE == "yes" ]
+then
+    $ROOT_DIR/bin/perf8 --memray -c $ROOT_DIR/bin/elastic-ingest --one-sync --sync-now --debug
+
+elif  [ $PERF_METRICS == "yes" ] 
+then
+    $ROOT_DIR/bin/perf8 --psutil -c $ROOT_DIR/bin/elastic-ingest --one-sync --sync-now --debug
```
Comment (tarekziade) [inline connectors/tests/ftest.sh:30]: we can now use:

```
$ROOT_DIR/bin/perf8 --memray --pyspy --psutil -c $ROOT_DIR/bin/elastic-ingest --one-sync --sync-now --debug
```

Comment (tarekziade) [reply connectors/tests/ftest.sh:30]: `--pyspy` gives you code profiling with no extra overhead. it works only on linux. it generates in the report dir a speedscope file yo can then load in https://speedscope.app

Comment (moxarth-elastic) [reply connectors/tests/ftest.sh]: Alright, I've updated the `ftest.sh` as per your suggestion 👍

**Review by tarekziade — APPROVED** [review]
LGTM (requires a rebase to land)

**Review by tarekziade — APPROVED** [review]

**Pull request merged**